src/cursor/cursor.ts

```diff
@@ -55,6 +57,83 @@ export interface CursorOptions extends CoreCursorOptions {
   numberOfRetries?: number;
 }
 
+export class CursorStream extends Readable {
+  [kCursor]: Cursor;
+
+  /** @event */
+  static readonly CLOSE = 'close' as const;
+  /** @event */
+  static readonly DATA = 'data' as const;
+  /** @event */
+  static readonly END = 'end' as const;
+  /** @event */
+  static readonly FINISH = 'finish' as const;
+  /** @event */
+  static readonly ERROR = 'error' as const;
+  /** @event */
+  static readonly PAUSE = 'pause' as const;
+  /** @event */
+  static readonly READABLE = 'readable' as const;
+  /** @event */
+  static readonly RESUME = 'resume' as const;
+
+  constructor(cursor: Cursor) {
+    super({ objectMode: true });
+    this[kCursor] = cursor;
+    this.on('end', () => this[kCursor].close(() => this.emit(Cursor.CLOSE)));
```

Comment:
should the `CursorStream` emit the `close` event still? I thought the `Cursor` would just emit this now? Seems to me that the stream should _only_ emit events that streams emit